altlas/Assets/scripts/MapGenerator.cs

```diff
@@ -1,65 +1,348 @@
 ﻿using System.Collections;
 using System.Collections.Generic;
+using System.Linq;
 using UnityEngine;
 
 public class MapGenerator : MonoBehaviour {
 
     public Transform map;
-    private float DRAWER_HOR_LEN = 0.36f;
-    private float DRAWER_VER_LEN = 0.44f;
+    private Loader loader;
+    private bool once = true;
 
-    List<MapClass> allMaps = new List<MapClass>();
+    //SCENE CONSTANTS FOR PLACING
+    private float DRAWER_HOR_LEN = 0.12f;
+    private float DRAWER_VER_LEN = 0.54f;
+    private float DRAWER_SHORT_X_DIS = 0.3f;
+    private float DRAWER_LONG_X_DIS = 1.4f;
+    private float DRAWER_Y_DIS = 0.24f;
 
+    private float DRAWER_Z_POSITION = 0.37f;
+    private float STARTING_DRAWER_X_POSITION = 0.70f;
+    private float STARTING_DRAWER_Y_POSITION = 0.6f;
+    private float MAPS_Y_OFFSET = 0.01f;
+
+    private Vector3 DESK_FREE_AREA_LEFT_CORNER = new Vector3(-0.7105434f, 0.85f, 0.4124395f);
+    private float DESK_FREE_AREA_LENGTH = 0.4f;
+
+    //List<MapData> allMaps = new List<MapData>();
+
+    Dictionary<string, Dictionary<string, List<MapData>>> subCats = new Dictionary<string, Dictionary<string, List<MapData>>>();
```

Comment:
Do we really need to save everything in a Dictionary? Would it not be easier to just iterate over every data object and determine the position on the fly without first storing them into a dictionary and then generate complex lookups?

Reply:
doing this makes it easier for me to spawn the categories in a drawer because i can just give the extracted list as a parameter to the spawnRowInDrawer function
you're right that this isn't the most efficient way, for your version we'd want some more functions and variables so we can use spawnMap in the loop and spawn it instantly to the corresponding drawer with rows still staying the same